File: tests/fixtures/grpc/monorepo/user-service/src/handler.go
package main

import (
	"context"
	pb "myproject/proto/user/v1"
)

type UserServiceServer struct {
	pb.UnimplementedUserServiceServer
}

func (s *UserServiceServer) GetUser(ctx context.Context, req *pb.GetUserRequest) (*pb.GetUserResponse, error) {
	return &pb.GetUserResponse{Id: req.Id, Name: "Alice"}, nil
}

func (s *UserServiceServer) CreateUser(ctx context.Context, req *pb.CreateUserRequest) (*pb.CreateUserResponse, error) {
	return &pb.CreateUserResponse{Id: "new-id"}, nil
}

File: tests/fixtures/grpc/multirepo/service-b/src/handler.java
package com.example.userservice;

import com.example.proto.UserServiceGrpc;
import com.example.proto.User.GetUserRequest;
import com.example.proto.User.GetUserResponse;
import com.example.proto.User.CreateUserRequest;
import com.example.proto.User.CreateUserResponse;

public class UserServiceImpl extends UserServiceGrpc.UserServiceImplBase {
    @Override
    public GetUserResponse getUser(GetUserRequest request) {
        return GetUserResponse.newBuilder()
            .setId(request.getId())
            .setName("Alice")
            .build();
    }

    @Override
    public CreateUserResponse createUser(CreateUserRequest request) {
        return CreateUserResponse.newBuilder()
            .setId("new-id")
            .build();
    }
}
